ETTrace/ETTrace/EMGPerfAnalysis.mm

```diff
@@ -24,158 +24,16 @@
 
 @implementation EMGPerfAnalysis
 
-static thread_t sMainMachThread = {0};
-static thread_t sETTraceThread = {0};
-
-static const int kMaxFramesPerStack = 512;
-static NSThread *sStackRecordingThread = nil;
-typedef struct {
-    CFTimeInterval time;
-    uint64_t frameCount;
-    uintptr_t frames[kMaxFramesPerStack];
-} Stack;
-
-typedef struct {
-    std::vector<Stack> *stacks;
-    char name[256];
-} Thread;
-static std::map<unsigned int, Thread *> *sThreadsMap;
-static std::mutex sThreadsLock;
-
 static dispatch_queue_t fileEventsQueue;
 
+static BOOL isRecording;
```

Comment:
nit: could this be a function inside `EMGTracer` which performs the original `sStackRecordingThread == null` check? I get a bit worried about state drifting apart over time.